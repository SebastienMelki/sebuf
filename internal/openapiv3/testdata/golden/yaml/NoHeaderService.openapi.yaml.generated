openapi: 3.1.0
info:
    title: NoHeaderService API
    version: 1.0.0
paths:
    /no-headers/simple:
        post:
            tags:
                - NoHeaderService
            summary: NoHeaders
            description: Method with no headers
            operationId: NoHeaders
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/HeaderRequest'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/HeaderResponse'
                "400":
                    description: Validation error
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/ValidationError'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
components:
    schemas:
        FieldViolation:
            type: object
            properties:
                field:
                    type: string
                    description: The field path that failed validation (e.g., 'user.email' for nested fields). For header validation, this will be the header name (e.g., 'X-API-Key')
                description:
                    type: string
                    description: Human-readable description of the validation violation (e.g., 'must be a valid email address', 'required field missing')
            required:
                - field
                - description
            description: FieldViolation describes a single validation error for a specific field.
        ValidationError:
            type: object
            properties:
                violations:
                    type: array
                    items:
                        $ref: '#/components/schemas/FieldViolation'
                    description: List of validation violations
            required:
                - violations
            description: ValidationError is returned when request validation fails. It contains a list of field violations describing what went wrong.
        HeaderRequest:
            type: object
            properties:
                data:
                    type: string
                    description: Request data
            description: Simple request message
        HeaderResponse:
            type: object
            properties:
                result:
                    type: string
                    description: Response data
            description: Simple response message
