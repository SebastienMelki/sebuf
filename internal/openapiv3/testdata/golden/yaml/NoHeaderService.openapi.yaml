openapi: 3.1.0
info:
    title: NoHeaderService API
    version: 1.0.0
paths:
    /no-headers/simple:
        post:
            tags:
                - NoHeaderService
            summary: NoHeaders
            description: Method with no headers
            operationId: NoHeaders
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/HeaderRequest'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/HeaderResponse'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
components:
    schemas:
        HeaderRequest:
            type: object
            properties:
                data:
                    type: string
                    description: Request data
            description: Simple request message
        HeaderResponse:
            type: object
            properties:
                result:
                    type: string
                    description: Response data
            description: Simple response message
